packages/alert-banner/src/AlertBanner.ts

```diff
@@ -0,0 +1,176 @@
+    html,
+    PropertyValues,
+    SpectrumElement,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { property } from '@spectrum-web-components/base/src/decorators.js';
+import '@spectrum-web-components/button/sp-close-button.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-alert.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-info.js';
+import styles from './alert-banner.css.js';
+
+const VALID_VARIANTS = ['neutral', 'info', 'negative'];
+export type AlertBannerVariants = typeof VALID_VARIANTS[number];
+
+/**
+ * @element sp-alert-banner
+ *
+ * @slot - The alert banner text context
+ * @slot action - Slot for the button element that surfaces the contextual action a user can take
+ *
+ * @fires close - Announces the alert banner has been closed
+ */
+export class AlertBanner extends SpectrumElement {
+    public static override get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    /**
+     * Controls the display of the alert banner
+     *
+     * @param {Boolean} open
+     */
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    /**
+     * Whether to include an icon-only close button to dismiss the alert banner
+     *
+     * @param {Boolean} dismissible
+     */
+    @property({ type: Boolean, reflect: true })
+    public dismissible = false;
+
+    /**
+     * The variant applies specific styling when set to `negative` or `info`;
+     * `variant` attribute is removed when it's passed an invalid variant.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String })
+    public set variant(variant: AlertBannerVariants) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+
+        if (this.isValidVariant(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+
+            if (window.__swc.DEBUG) {
+                window.__swc.warn(
+                    this,
+                    `<${this.localName}> element expects the "variant" attribute to be one of the following:`,
+                    'https://opensource.adobe.com/spectrum-web-components/components/alert-banner/#variants',
+                    {
+                        issues: [...VALID_VARIANTS],
+                    }
+                );
+            }
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): AlertBannerVariants {
+        return this._variant;
+    }
+
+    private _variant: AlertBannerVariants = '';
+
+    protected isValidVariant(variant: string): boolean {
+        return VALID_VARIANTS.includes(variant);
+    }
+
+    protected renderIcon(variant: string): TemplateResult {
+        switch (variant) {
+            case 'info':
+                return html`
+                    <sp-icon-info
```

Comment:
Is this not customisable?

Reply:
No, in the [RFC](https://github.com/adobe/spectrum-web-components/discussions/4106) we didn't discuss about making the icon customizable and I see that the similar `Toast` component doesn't let you customize the icon either